src/RA_StringUtils.h

```diff
@@ -441,49 +453,49 @@ class StringBuilder
             CharRef,
             WCharRef,
         };
-        Type DataType{ Type::String };
+        Type DataType{Type::String};
     };
 
     mutable std::vector<PendingString> m_vPending;
-};
+}; // namespace ra
 
 template<typename... Ts>
-void AppendString(std::string& sString, Ts... args)
+void AppendString(std::string& sString, const Ts... args)
 {
     StringBuilder builder;
-    builder.Append(std::forward<Ts>(args)...);
+    builder.Append(std::forward<const Ts>(args)...);
     builder.AppendToString(sString);
 }
 
 template<typename... Ts>
-void AppendWString(std::wstring& sString, Ts... args)
+void AppendWString(std::wstring& sString, const Ts... args)
 {
     StringBuilder builder(true);
-    builder.Append(std::forward<Ts>(args)...);
+    builder.Append(std::forward<const Ts>(args)...);
     builder.AppendToWString(sString);
 }
 
 template<typename... Ts>
-std::string BuildString(Ts... args)
+std::string BuildString(const Ts... args)
 {
     StringBuilder builder;
-    builder.Append(std::forward<Ts>(args)...);
+    builder.Append(std::forward<const Ts>(args)...);
     return builder.ToString();
 }
 
 template<typename... Ts>
-std::wstring BuildWString(Ts... args)
+std::wstring BuildWString(const Ts... args)
 {
     StringBuilder builder(true);
-    builder.Append(std::forward<Ts>(args)...);
+    builder.Append(std::forward<const Ts>(args)...);
     return builder.ToWString();
 }
 
 /// <summary>
 /// Constructs a <see cref="std::basic_string" /> from a format string and parameters
 /// </summary>
 template<typename CharT, typename = std::enable_if_t<is_char_v<CharT>>, typename... Ts>
-_NODISCARD inline auto StringPrintf(_In_z_ _Printf_format_string_ const CharT* const __restrict sFormat, Ts... args)
+_NODISCARD inline auto StringPrintf(_In_z_ _Printf_format_string_ const CharT* const __restrict sFormat, Ts&&... args)
```

Comment:
I'm not 100% sure. But I think last you said it didn't work because it couldn't forward the const references. 
I believe it's better to get rid of the && here and got rid of the consts for the other parameter packs.

I'd have to test it myself to be sure.

Reply:
I believe that's related to the string literals. The input `Ts` must be non-const, but through forwarding magic, they can be converted to `const`.